aidial_adapter_bedrock/llm/model/claude/v3/tools.py

```diff
@@ -0,0 +1,47 @@
+from enum import Enum
+from typing import assert_never
+from aidial_sdk.chat_completion import FunctionCall, ToolCall
+from anthropic.types import ToolUseBlock
+
+
+import json
+
+from aidial_adapter_bedrock.llm.consumer import Consumer
+
+
+def to_dial_tool_call(block: ToolUseBlock) -> ToolCall:
+    return ToolCall(
+        index=None,
+        id=block.id,
+        type="function",
+        function=FunctionCall(
+            name=block.name,
+            arguments=json.dumps(block.input),
+        ),
+    )
+
+
+def to_dial_function_call(block: ToolUseBlock) -> FunctionCall:
+    return FunctionCall(name=block.name, arguments=json.dumps(block.input))
+
+
+class ToolsMode(Enum):
+    NATIVE_TOOLS = "NATIVE_TOOLS"
+    """
+    Claude V3 API Supports only tools.
+    But our API supports also deprecated "functions" which are pretty same as tools,
+    so we can emulate them.
+    """
+    FUNCTION_EMULATION = "FUNCTION_EMULATION"
```

Comment:
`FUNCTION_EMULATION` is misleading. There is nothing we emulate here. 
It's just two different ways to convert Claude tool usage to OpenAI - OpenAI functions or OpenAI tools.

I suggest to rename:

```python
class OpenAIToolsMode(Enum):
    TOOLS = "TOOLS"
    FUNCTIONS = "FUNCTIONS"
```

Reply:
Well, I disagree on that

Claude doesn't support `Functions`, but support `Tools`

Since we want to allow user to use Functions in his request, adapter should make this convertions

1. First call:
`User` -> `Request with Functions` -> `Adapter` -> `| Converts Functions into Tools |` -> `Model`

2. Response from model:
`Model` -> `{Tool Call Message}` -> `Adapter` -> `|Converts Tool Call Message to Function Call Message|` -> User

3. User response with function result:

`User` -> `Function Call Message + Function Result Message` -> `Adapter` -> `|Converts both call and result from Function to Message format|` -> `Model`

---

I can rename it to something like `FUNCTION_ADAPTER`, but it's just make some naming collisions with ModelAdapter

But I definetly think we need some modification to `*_FUNCTIONS`, becasue it's not common behaviour

We modify both input to model, and output from model

Reply:
Fixed naming